src/impl/formatter.js

```diff
@@ -375,29 +375,30 @@ export default class Formatter {
     const tokenToField = (token) => {
         switch (token[0]) {
           case "S":
-            return "millisecond";
+            return "milliseconds";
           case "s":
-            return "second";
+            return "seconds";
           case "m":
-            return "minute";
+            return "minutes";
           case "h":
-            return "hour";
+            return "hours";
           case "d":
-            return "day";
+            return "days";
           case "w":
-            return "week";
+            return "weeks";
           case "M":
-            return "month";
+            return "months";
           case "y":
-            return "year";
+            return "years";
           default:
             return null;
         }
       },
       tokenToString = (lildur) => (token) => {
         const mapped = tokenToField(token);
         if (mapped) {
-          return this.num(lildur.get(mapped), token.length);
+          const inversionFactor = lildur < 0 && mapped !== Object.keys(lildur.values)[0] ? -1 : 1;
```

Comment:
1. We should not rely on the order of keys in `lildur.values`
2. `lildur < 0` converts the duration to milliseconds every time. This should be done just once.

Reply:
Not sure if there is an _easy_ way. We'd have to find the index in `orderedUnits`. `shiftTo` happens here every time, can you check if that has a defined key order? But even if it does, it feels wrong to rely on that.

Reply:
[duration.js line 794](https://github.com/dobon/luxon/blob/a6c5577966f5f071343f0fe35da82b64c09189a4/src/duration.js#L794C5-L821C6)

`shiftTo` builds up `values` by looping through `orderedUnits`.